lib/util/locale.ts

```diff
@@ -0,0 +1,28 @@
+import { IntlShape } from 'react-intl'
+import { MouseEvent } from 'react'
+import { User } from '@auth0/auth0-react'
+
+export const handleLocaleSelection = (
+  e: MouseEvent<Element>,
+  locale: string,
+  currentLocale: string,
+  loggedInUser: User,
+  createOrUpdateUser: (user: User, silent: boolean, intl: IntlShape) => void,
+  setLocale: (locale: string) => void,
```

Comment:
Portions of the code for changing the locale should be a redux action (`actions/ui`), so it can take advantage of the redux state so that we don't have to pass all of these arguments around.

Reply:
Good call! Turns out most of the wrapper stuff wasn't even needed! Things are much cleaner now get excited